worker/peergrouper/desired.go

```diff
@@ -220,23 +224,26 @@ func addNewMembers(
 	mongoSpace network.SpaceName,
 ) {
 	for _, m := range toKeep {
+		if members[m] != nil {
+			continue
+		}
 		hasAddress := m.SelectMongoHostPort(mongoSpace) != ""
-		if members[m] == nil && hasAddress {
```

Comment:
the old code was doing multiple SelectMongoHostPort calls for existing members.
If the member already exists, then we don't need to know if it hasAddress because its members[m] won't be nil.